Result/ResultType.swift

```diff
@@ -64,6 +64,30 @@ public extension ResultType {
 	}
 }
 
+/// Protocol used to constrain a version of `map` for `transform`s that throw.
+public protocol ErrorTypeConvertible: ErrorType {
+	typealias ConvertibleType = Self
```

Comment:
It is necessary because NSError is not final. From what I've seen, this is a common pattern.